runtime/shared_heap.c

```diff
@@ -612,6 +597,52 @@ void caml_redarken_pool(struct pool* r, scanning_action f, void* fdata) {
 }
 
 
+/* Heap and freelist stats */
+
+/* Move the given heap stats to the orphan pools.
+   You need to hold the [pool_freelist] lock. */
+static void orphan_heap_stats_with_lock(struct caml_heap_state *heap) {
+  caml_accum_heap_stats(&pool_freelist.stats, &heap->stats);
+  memset(&heap->stats, 0, sizeof(heap->stats));
+}
+
+/* The stats for an adopted pool are moved from the free pool stats to
+   the heap stats of the adopting domain.
+   You need to hold the [pool_freelist] lock. */
+static void adopt_pool_stats_with_lock(
+  struct caml_heap_state* adopter, pool *r, sizeclass sz)
+{
+    struct heap_stats pool_stats = { 0 };
```

Comment:
I think I mentioned yesterday that this was the role of the trailing comma in another piece of code, but in fact reading again I think I was wrong: `{ 0 }` and `{ 0, }` are equivalent. (Which is good news, because this `{ 0 }` has been here a long time and it would be a bug.) I'll still use `{ 0, }` for consistency, thanks!

(Reference: [How to initialise structures to all-elements-zero-or-null](https://www.ex-parrot.com/~chris/random/initialise.html).)

Reply:
Right, in this scenario my understanding of an initializer list is that `{ 0 }` means "set the first member to zero (and implicitly everything else to zero)" and ` { 0, }` is functionally equivalent.

Where it differ would be using ` { 0xFF, }` where it would mean "set the first member to 0xFF, and then implicitly the rest to zero" ?
Am I getting it right?

Reply:
Note quite. `{ 0xFF, }` and `{ 0xFF }` are entirely equivalent, and will initialize everything but the first member to `0`. There is just no way to default-initialize all the members of an aggregate to something different than 0 in C.